src/triage/component/postmodeling/utils/predictions_selected_model_groups.py

```diff
@@ -0,0 +1,202 @@
+# for the relevant models. 
+# This util is primarily targetted to be used post-audition for the following use-case:
+# 1. A large model grid was used in the experiment with (save_predictions=False)
+# 2. Audition was used to narrowdown model_groups of interest & predictions are required for postmodeling
+
+
+def _fetch_relevant_model_matrix_info(db_engine, experiment_hash, model_groups):
+    """ For the given experiment and model groups, fetch the model_ids, and match them with their train/test matrix pairs 
+        Args:
+            experiment_hash (str): The experiment hash we are interested in
+            model_groups (List[int]): A list of model groups in the experiment
+
+        Return: 
+            A DataFrame that contains the relevant model_ids, train_end_times, and matrix information
+    """
+
+    q = """
+        SELECT 
+            model_id,
+            train_end_time,
+            model_hash,
+            model_group_id,
+            train_matrix_uuid,
+            c.matrix_uuid as test_matrix_uuid
+        FROM triage_metadata.experiment_models a 
+            JOIN triage_metadata.models b using(model_hash)
+                JOIN test_results.prediction_metadata c using(model_id)
+        WHERE experiment_hash='{experiment_hash}' AND model_group_id in {model_groups}
+    """.format(
+        experiment_hash=experiment_hash,
+        model_groups=tuple(model_groups)
+    )
+
+    return pd.read_sql(q, db_engine)
+
+    
+def generate_predictions(db_engine, experiment_hash, model_groups, project_path, range_train_end_times=None, rank_odrer='worst', replace=False):
+    """ Generate predictions and write to DB for a set of model_groups in an experiment
+        Args:
+            db_conn: Sqlalchemy engine
+            experiment_hash (str): hash of the experiment (currently handling only one experiment)
+            model_groups (list): The list of model group ids we are interested in (ideally, chosen through audition)
+            project_path (str): Path where the created matrices and trained model objects are stored for the experiment
+            range_train_end_times (List): Optional. If given, only the models with train_end_times that fall in the range are scored.
+                                        Should be a list of two dates (str). [range_start_date, range_end_date]
+            rank_order (str) : How to deal with ties in the scores. 
+            replace (bool) : Whether to overwrite the preditctions for a model_id, if already found in the DB
+
+        Returns: None
+            This directly writes to the test_results.predictions table
+    """
+    model_matrix_info = _fetch_relevant_model_matrix_info(
+        db_engine=db_engine,
+        experiment_hash=experiment_hash,
+        model_groups=model_groups
+    )
+
+    if len(model_matrix_info)==0:
+        raise ValueError('No models were found for the given experiment and model group(s)')
```

Comment:
might want to at least generate a warning (or error) if there were any model groups for which no models were found (not just overall)

Reply:
Yeah. That's a good point. Set it up so that it will raise an error if any of the model groups are not present. To make sure that the choices in the config file exactly match the saved predictions.